var/spack/repos/builtin/packages/py-pymol/package.py

```diff
@@ -13,20 +13,22 @@ class PyPymol(PythonPackage):
        movies. Open Source PyMOL is free to everyone!"""
 
     homepage = "https://pymol.org"
-    url      = "https://sourceforge.net/projects/pymol/files/pymol/2/pymol-v2.1.0.tar.bz2"
+    url      = "https://github.com/schrodinger/pymol-open-source/archive/v2.3.0.tar.gz"
 
-    version('2.1.0', sha256='7ae8ebb899533d691a67c1ec731b00518dea456ab3e258aa052a65c24b63eae2')
+    version('2.3.0', sha256='62aa21fafd1db805c876f89466e47513809f8198395e1f00a5f5cc40d6f40ed0')
 
     depends_on('python+tkinter', type=('build', 'run'))
-    depends_on('tcl')
-    depends_on('tk')
-    depends_on('py-pmw')
-    depends_on('gl')
-    depends_on('glu')
+    depends_on('freetype', type=('build', 'run'))
     depends_on('glew')
-    depends_on('libpng')
-    depends_on('freetype')
-    depends_on('libxml2')
-    depends_on('msgpack-c')
-    depends_on('py-pyqt4', type=('build', 'run'))
-    depends_on('freeglut')
+    depends_on('glm')
+    depends_on('freeglut', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('tcsh', type=('build', 'run'))
+    depends_on('py-pyqt5', type=('build', 'run'))
+    depends_on('py-pmw', type=('build', 'run'))
+    depends_on('mesa-glu', type=('build', 'run'))
+
+    depends_on('libmmtf-cpp', type=('build'))
+    depends_on('msgpack-c', type=('build'))
+
+    patch('apbstools_tcltk8.6.patch', level=0, when='@2.3.0')
```

Comment:
Can you explain this patch in more detail? Why is it necessary? Has it been submitted upstream?

Reply:
@adamjstewart I just copied the patch from the archlinux pymol installation. But a more careful inspection makes me believe that it is not needed. First, the plugin it tries to fix is outdated.  Also, we do not provide apbs support at this moment, so it is useless.